bugtool/cmd/root.go

```diff
@@ -347,6 +353,30 @@ func runAll(commands []string, cmdDir string, k8sPods []string) {
 	}
 }
 
+func removeObjectFiles(cmdDir string, k8sPods []string) {
+	// Remove object files for each endpoint. Endpoints directories are in the
+	// state directory and have numerical names.
+	rmFunc := func(path string) {
+		rmCmd := fmt.Sprintf("rm -rf %s", filepath.Join(path, "[0-9]*", "*.o"))
+		// The exec package doesn't expand glob patterns, call the shell instead
+		cmd := exec.Command("bash", "-c", rmCmd)
+		err := cmd.Run()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to exclude object files: %s\n", err)
+		}
+	}
+
+	if len(k8sPods) > 0 {
+		for _, pod := range k8sPods {
+			path := filepath.Join(cmdDir, fmt.Sprintf("%s-%s", pod, defaults.StateDir))
+			rmFunc(path)
+		}
```

Comment:
Hey @pchaigno, thanks for the feedback.
My current understanding of bugtool is that it can be used directly from a cilium-agent pod or from a remote machine, using the `--k8s-mode` flag.
I'm still pretty new to the project so I might be overlooking something obvious about the use of `cilium-bugtool`. :smile: 
Please let me know if I'm missing something here, I'll adapt the PR.

During my tests I've observed that when I run `cilium-butool` from within a cilium agent pod, the archive is structured like this:
```
root@kubeadm01:/tmp/cilium-bugtool-20221202-101329.136+0000-UTC-3580203555# tree -d
.
`-- cmd
    `-- state
        |-- 12
        |-- 2351
        |-- 29
        |-- 4041
        |-- globals
        `-- templates
            |-- 99e182ec920f56005679e3b9e405b875cca519b2fbb6f989a9d5be76eced7cb0
            `-- f9e440c3fd81b0a9e3154d555c4ddb6f467a6c38db8ed3cf39e6a33f5b50af48
```

When I run it from my laptop `cilium-bugtool --k8s-mode`, the archive is structured like this (a state dir per pod):
```
$ tree -d
.
└── cmd
    ├── cilium-jjxnw-state
    │   ├── 12
    │   ├── 2351
    │   ├── 29
    │   ├── 4041
    │   ├── globals
    │   └── templates
    │       ├── 99e182ec920f56005679e3b9e405b875cca519b2fbb6f989a9d5be76eced7cb0
    │       └── f9e440c3fd81b0a9e3154d555c4ddb6f467a6c38db8ed3cf39e6a33f5b50af48
    └── cilium-wws24-state
        ├── 1084
        ├── 176
        ├── 2099
        ├── 2334
        ├── 2446
        ├── 473
        ├── 795
        ├── globals
        └── templates
            ├── 2472c3787e7512efa300c83156ddf05f0b05f236d6e504d6266f727a03ba6bfd
            └── b68645ca2ba2323f59c0b2fc34cd21a418b9c09b54305f10d9b4a510f99cd1f5
```

Does that make sense?
Thanks!

Reply:
You might be new to the project, but TIL about `--k8s-mode` :smile: 

Maybe we should however check on that specific flag instead of checking the number of Cilium pods? They are the same, but it sounds like what we intend to check is the mode in which we run; the number of pods is just an indirect way to get that.

Reply:
@pchaigno Ahah yeah, with such a large codebase, it's not surprising. It happens to me all the time :wink: 

Makes sense to me, yes.
There are other checks for `len(k8sPods)` elsewhere in the code (e.g: [here](https://github.com/cilium/cilium/blob/79c6f5725372b52c9877a9bde1249da039948649/bugtool/cmd/configuration.go#L321) or [here](https://github.com/cilium/cilium/blob/79c6f5725372b52c9877a9bde1249da039948649/bugtool/cmd/configuration.go#L413)) so I assumed it was the correct way to express the condition.
But after looking a bit more at the code, it looks like the only way to get `k8sPods` is when [the flag is set](https://github.com/cilium/cilium/blob/79c6f5725372b52c9877a9bde1249da039948649/bugtool/cmd/root.go#L108). Using the flag makes the intent a bit clearer.

Reply:
> There are other checks for `len(k8sPods)` elsewhere in the code (e.g: [here](https://github.com/cilium/cilium/blob/79c6f5725372b52c9877a9bde1249da039948649/bugtool/cmd/configuration.go#L321) or [here](https://github.com/cilium/cilium/blob/79c6f5725372b52c9877a9bde1249da039948649/bugtool/cmd/configuration.go#L413)) so I assumed it was the correct way to express the condition.

IMO, we should also fix those other places, but that's orthogonal to your changes.